src/plugins/filter/filter.js

```diff
@@ -99,6 +104,82 @@ var componentName = "wb-filter",
 			replace( /_NBITEM_/g, nbItem ).
 			replace( /_TOTAL_/g, total );
 	},
+
+	/*
+	 * Takes in the text from the filter box
+	 * Returns:
+	 *  An array of search words
+	 *      Special characters are escaped
+	 *      Double and single quotes removed
+	 */
+	filterQueryParser = function( filter ) {
+
+		// Pattern to seperate the filter text into "words"
+		var pattern = /[^\s"]+|"([^"]*)"/gi;
+
+		// Make strings safe again for regex
+		// Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+		filter = filter.replace( /[.*+?^${}()|[\]\\]/g, "\\$&" );
+
+		// Apply the word match pattern and return
+		return filter.match( pattern );
+	},
+
+	/**
+	 * Build the Regular Expression that we are going
+	 * to use to filter content
+	 * This involves identifying the type of search being
+	 * applied and then breaking up the search string into
+	 * words
+	 * We are using object literals to increase readability
+	 * and reduce the amount of case / if statements we need
+	 */
+	buildSearchFilterRegularExp =  function( filterType, filter ) {
+
+		var filters, searchFilterRegExp, searchFilterWords;
+
+		// Build regular expressions for type = exact
+		function isExact() {
+			searchFilterRegExp = new RegExp( filter, "i" );
+			return searchFilterRegExp;
+		}
+
+		// Build regular expressions for type = and
+		function isAND() {
+			searchFilterWords = filterQueryParser( filter );
+			if ( searchFilterWords ) {
+				var andSearchFilterString = ".*";
+				for ( var i = 0; i < searchFilterWords.length; i++ ) {
+					andSearchFilterString += ( "(?=.*" + searchFilterWords[ i ] + ")" );
+				}
+				searchFilterRegExp = new RegExp( andSearchFilterString, "gi" );
+			} else {
+				searchFilterRegExp = new RegExp( "", "gi" );
+			}
+			return searchFilterRegExp;
+		}
+
+		// Build regular expressions for type = or
+		function isOR() {
+			searchFilterWords = filterQueryParser( filter );
+			searchFilterRegExp = ( searchFilterWords ) ? new RegExp( searchFilterWords.join( "|" ), "gi" ) : new RegExp( "", "gi" );
+			return searchFilterRegExp;
+		}
+
+		var filters = {
+			"default": isExact,
+			"exact": isExact,
+			"EXACT": isExact,
+			"and": isAND,
+			"AND": isAND,
+			"or": isOR,
+			"OR": isOR
+		};
+
+		// Check if the provided filterType value is valid and then run or use default filterTyep
+		return ( filters.hasOwnProperty( filterType ) ? filters[ filterType ]() : filters.default() );
```

Comment:
``` filters.hasOwnProperty( filterType ) ``` try this ``` filters.hasOwnProperty( filterType.toLowerCase() ) ```
This will avoid the duplicate in properties in the filter object